### PR TITLE
[APB-3367][MP] Add feature flag for agent-led-deauth

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/FrontendModule.scala [hunk 1]
```diff
@@ -101,6 +101,7 @@ class FrontendModule(val environment: Environment, val configuration: Configurat
     bindBooleanProperty("features.redirect-to-confirm-personal-income")
     bindBooleanProperty("features.redirect-to-confirm-mtd-it")
     bindBooleanProperty("features.redirect-to-confirm-mtd-vat")
+    bindBooleanProperty("features.show-agent-led-de-auth")
```
Comment (sureshhmrc) [inline app/FrontendModule.scala]: i think we should first disable the flag in the `agent-services-account-frontend`

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala [hunk 2]
```diff
@@ -75,62 +75,72 @@ class AgentLedDeAuthController @Inject()(
...
   }
 
   def submitClientType: Action[AnyContent] = Action.async { implicit request =>
-    handleSubmitClientType
+    ifShowDeAuthFlag(handleSubmitClientType)
   }
 
   def showSelectService: Action[AnyContent] = Action.async { implicit request =>
-    handleGetSelectServicePage(agentConfirmationForm("cancel-authorisation.error.business-service.required"))
+    ifShowDeAuthFlag(
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: i guess we don't need this logic to all endpoints,  we just need for `/client-type`

**Comment by sureshhmrc** [comment]
also have you thought about commenting out the

File: app/FrontendModule.scala (hunk 1 above)
Comment (mariannepearson0) [reply app/FrontendModule.scala]: 👍

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala (hunk 2 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: Could they not navigate to the other pages?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: even if they manually navigate to other pages, they will get redirected to /client-type due to client-type missing cache

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala [hunk 3]
```diff
@@ -75,62 +75,72 @@ class AgentLedDeAuthController @Inject()(
     Redirect(agentsLedDeAuthRootUrl)
   }
 
+  private def ifShowDeAuthFlag(result: Future[Result]) =
+    if (featureFlags.showAgentLedDeAuth) result
+    else {
+      Logger(getClass).warn("Agent led de authorisation feature is disabled.")
+      Future successful Unauthorized
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: `Unauthorized` is misleading - it should be something like notSupported or notImplemented

File: app/FrontendModule.scala (hunk 1 above)
Comment (mariannepearson0) [reply app/FrontendModule.scala]: So there is a flag for this link in agent-services-account-frontend, shall i just turn that off

Comment (sureshhmrc) [reply app/FrontendModule.scala]: yes, i added this yesterday 
`https://github.com/hmrc/agent-services-account-frontend/blob/master/conf/application.conf#L169`

**Pull request merged**